In [12]:
import pandas as pd
import plotly.graph_objects as go
import numpy as np
import matplotlib.pyplot as plt
from matplotlib.pyplot import figure
import datetime
import plotly.express as px
import math
import warnings
warnings.filterwarnings('ignore')
import time
from time import perf_counter
import glob
import tweets_utils
import csv
import importlib
import seaborn as sns
path = glob.glob("C:/Users/gianl/Desktop/Gi/Supsi/BachelorProject/")
%autosave 3600

Autosaving every 3600 seconds


## Indice
- [Check if the users are inside current data](#check)
- [Collecting data and create DataFrame](#collecting_350)

In [13]:
# x = pd.DataFrame(x, columns=["screen_name"])
# x.to_csv("C:/Users/gianl/Desktop/Gi/Supsi/BachelorProject/csv/300_good_users.csv", 
#          line_terminator="\n", index=False, encoding="utf-8")

In [15]:
df = pd.read_csv(path[0] + "/csv/300_good_users.csv", low_memory=False, lineterminator="\n")

In [28]:
df

screen_name
0    HelenBranswell
1         mlipsitch
2             trvrb
3            kakape
4      JeremyFarrar
..              ...
345    greg_folkers
346         ISARIC1
347        WHOGOARN
348      jmayer0716
349     robyn_s_lee

[350 rows x 1 columns]

# Check if 350 good users are inside current data
<a id="check"/>

In [21]:
files_tweets = glob.glob("F:/Supsi/COVID-19-Tweets/*/tweets.csv")
user_screen_name = []
cols_tweets = [u'user_screen_name']

In [23]:
result = []
importlib.reload(tweets_utils)
start_time = time.perf_counter()
j = 0
for filename in files_tweets:
    print(f"Reading file: {filename}")
    result.extend(tweets_utils.process_all_data(filename, cols_tweets, True, list(df["screen_name"])))
    lista = result[j]
    j = j + 1
    print(f"Mese numero {j}")
    for i in range(len(lista)):

        user_screen_name.extend(lista[i]["users"])
stop_time = time.perf_counter()
print("Time: ",stop_time-start_time)

Reading file: F:/Supsi/COVID-19-Tweets\2020-01\tweets.csv
Processing chunk 1
Processing chunk 2
Processing chunk 3
Processing chunk 4
Processing chunk 5
Mese numero 1
Reading file: F:/Supsi/COVID-19-Tweets\2020-02\tweets.csv
Processing chunk 1
Processing chunk 2
Processing chunk 3
Processing chunk 4
Processing chunk 5
Processing chunk 6
Processing chunk 7
Processing chunk 8
Processing chunk 9
Processing chunk 10
Processing chunk 11
Processing chunk 12
Mese numero 2
Reading file: F:/Supsi/COVID-19-Tweets\2020-03\tweets.csv
Processing chunk 1
Processing chunk 2
Processing chunk 3
Processing chunk 4
Processing chunk 5
Processing chunk 6
Processing chunk 7
Processing chunk 8
Processing chunk 9
Processing chunk 10
Processing chunk 11
Processing chunk 12
Processing chunk 13
Processing chunk 14
Processing chunk 15
Mese numero 3
Reading file: F:/Supsi/COVID-19-Tweets\2020-04\tweets.csv
Processing chunk 1
Processing chunk 2
Processing chunk 3
Processing chunk 4
Processing chunk 5
Processing chu

In [34]:
lst_found = list(set(user_screen_name))
val_350 = len(df["screen_name"])
val_found = len(user_screen_name)
print(f"{round(val_found/val_350,2)}%")
print(f"presenti in {len(lst_found)}")

93.66%
presenti in 338


In [38]:
diff = list(set(df["screen_name"]) - set(lst_found))
diff

['POTUS',
 'JInterlandi',
 'WHCOVIDResponse',
 'dmaccannell',
 'WHCOS',
 'ZekeEmanuel',
 'marcelsalathe',
 'CarolineColijn',
 'UKAMREnvoy',
 'JeffDSachs',
 'BillGates',
 'Declan_M_Butler']

# Collecting data
<a id="collecting_350" />

In [46]:
cols = [
    u'id',u'user_id',u'user_screen_name',u'urls', u'hashtags',
    u'rt_user_screen_name', u'rt_user_id', 
    u'in_reply_to_screen_name', u'in_reply_to_user_id', 'created_at', 'retweet_count']

In [47]:
result = []
data = pd.DataFrame()
importlib.reload(tweets_utils)
start_time = time.perf_counter()
j = 0
for filename in files_tweets:
    print(f"Reading file: {filename}")
    result.extend(tweets_utils.process_all_data(filename, cols, True, df["screen_name"]))
    lista = result[j]
    j = j + 1
    print(f"Mese numero {j}")
    for i in range(len(lista)):
        data = data.append(lista[i]["df"])
    print(len(data))
stop_time = time.perf_counter()
print("Time: ",stop_time-start_time)

Reading file: F:/Supsi/COVID-19-Tweets\2020-01\tweets.csv
Processing chunk 1
Processing chunk 2
Processing chunk 3
Processing chunk 4
Processing chunk 5
Mese numero 1
6614
Reading file: F:/Supsi/COVID-19-Tweets\2020-02\tweets.csv
Processing chunk 1
Processing chunk 2
Processing chunk 3
Processing chunk 4
Processing chunk 5
Processing chunk 6
Processing chunk 7
Processing chunk 8
Processing chunk 9
Processing chunk 10
Processing chunk 11
Processing chunk 12
Mese numero 2
17300
Reading file: F:/Supsi/COVID-19-Tweets\2020-03\tweets.csv
Processing chunk 1
Processing chunk 2
Processing chunk 3
Processing chunk 4
Processing chunk 5
Processing chunk 6
Processing chunk 7
Processing chunk 8
Processing chunk 9
Processing chunk 10
Processing chunk 11
Processing chunk 12
Processing chunk 13
Processing chunk 14
Processing chunk 15
Mese numero 3
24031
Reading file: F:/Supsi/COVID-19-Tweets\2020-04\tweets.csv
Processing chunk 1
Processing chunk 2
Processing chunk 3
Processing chunk 4
Processing chunk

In [45]:
data

user_screen_name
182              MackayIM
365       KindrachukJason
1413             WHOGOARN
1516             MackayIM
1879             MackayIM
...                   ...
23401321          nytimes
23402249    SaskiaPopescu
23403719     DanLarremore
23404328         stgoldst
23411172     EvolveDotZoo

[32781 rows x 1 columns]

In [49]:
data = data.reset_index(drop = True)
data

id                      created_at       user_id  \
0      1219783336081539073  Wed Jan 22 00:46:24 +0000 2020  1.352962e+09   
1      1219810588873961473  Wed Jan 22 02:34:42 +0000 2020  1.071979e+18   
2      1219903528967249925  Wed Jan 22 08:44:00 +0000 2020  3.528640e+08   
3      1219915043896119296  Wed Jan 22 09:29:46 +0000 2020  1.352962e+09   
4      1219915530355666944  Wed Jan 22 09:31:42 +0000 2020  1.352962e+09   
...                    ...                             ...           ...   
32776  1260399384073732098  Wed May 13 02:40:04 +0000 2020  8.070950e+05   
32777  1261351389818585088  Fri May 15 17:43:00 +0000 2020  2.611794e+07   
32778  1265460966126485506  Wed May 27 01:53:00 +0000 2020  1.979288e+09   
32779  1256818060050067460  Sun May 03 05:29:10 +0000 2020  1.245975e+09   
32780  1257237738350403585  Mon May 04 09:16:49 +0000 2020  3.061632e+09   

      user_screen_name  retweet_count  in_reply_to_user_id  \
0             MackayIM           16.0                  NaN   
1      KindrachukJason           20.0                  NaN   
2             WHOGOARN          542.0                  NaN   
3             MackayIM            0.0                  NaN   
4             MackayIM            2.0                  NaN   
...                ...            ...                  ...   
32776          nytimes          191.0                  NaN   
32777    SaskiaPopescu           37.0                  NaN   
32778     DanLarremore            1.0                  NaN   
32779         stgoldst            1.0                  NaN   
32780     EvolveDotZoo           71.0                  NaN   

      in_reply_to_screen_name    rt_user_id rt_user_screen_name  \
0                         NaN           NaN                 NaN   
1                         NaN  7.641714e+07            IDSAInfo   
2                         NaN  1.898686e+08            DrTedros   
3                         NaN           NaN                 NaN   
4                         NaN           NaN                 NaN   
...                       ...           ...                 ...   
32776                     NaN           NaN                 NaN   
32777                     NaN  2.880886e+09        gregkoblentz   
32778                     NaN  3.967445e+07              neurdy   
32779                     NaN           NaN                 NaN   
32780                     NaN  1.159777e+18       OscarMacLean1   

                                                hashtags  \
0      [{'text': 'Wuhan', 'indices': [87, 93]}, {'tex...   
1      [{'text': 'Wuhan', 'indices': [64, 70]}, {'tex...   
2         [{'text': 'coronavirus', 'indices': [83, 95]}]   
3      [{'text': 'WuhanPneumonia', 'indices': [91, 10...   
4                                                     []   
...                                                  ...   
32776                                                 []   
32777                                                 []   
32778  [{'text': 'COVID19colorado', 'indices': [20, 3...   
32779                                                 []   
32780                                                 []   

                                                    urls  
0      [{'url': 'https://t.co/fbk6HGY3gb', 'expanded_...  
1      [{'url': 'https://t.co/Cm8rjaZHpi', 'expanded_...  
2                                                     []  
3      [{'url': 'https://t.co/eTMXY7oH2O', 'expanded_...  
4      [{'url': 'https://t.co/187Fuuj01y', 'expanded_...  
...                                                  ...  
32776  [{'url': 'https://t.co/aOB2mL20Js', 'expanded_...  
32777  [{'url': 'https://t.co/LiE05sOjW5', 'expanded_...  
32778                                                 []  
32779  [{'url': 'https://t.co/25pu5b4qoy', 'expanded_...  
32780                                                 []  

[32781 rows x 11 columns]

In [51]:
# data = data.reset_index(drop = True)
path = glob.glob("C:/Users/gianl/Desktop/Gi/Supsi/BachelorProject/large_files/")
data.to_csv(path[0] + "all_good_users.csv", index = False, line_terminator = "\n", encoding = "utf-8")